pkg/app/piped/appconfigreporter/appconfigreporter.go

```diff
@@ -309,10 +242,61 @@ func (r *Reporter) updateUnregisteredApps(ctx context.Context, registeredAppPath
 	return nil
 }
 
+// findRegisteredApps finds out registered application info that should be updated in the given git repository.
+func (r *Reporter) findRegisteredApps(repoPath, repoID string) ([]*model.ApplicationInfo, error) {
+	// Compare the apps registered on Control-plane with the latest config file
+	// and return only the ones that have been changed.
+	apps := make([]*model.ApplicationInfo, 0)
+	for _, app := range r.applicationLister.List() {
+		if app.GitPath.Repo.Id != repoID {
+			continue
+		}
+		appCfg, err := r.readApplicationInfo(repoPath, repoID, app.GitPath.Path, app.GitPath.ConfigFilename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read application config file: %w", err)
+		}
+		if isSynced(appCfg, app) {
+			continue
+		}
+		appCfg.Id = app.Id
+		apps = append(apps, appCfg)
+	}
+	return apps, nil
+}
+
+func isSynced(appInfo *model.ApplicationInfo, app *model.Application) bool {
+	// TODO: Make it possible to follow the ApplicationInfo field changes
+	if appInfo.Name != app.Name {
+		return false
+	}
+	if appInfo.Kind != app.Kind {
+		return false
+	}
```

Comment:
Changing the Kind could cause unexpected behavior since the application is paired with a Cloud Provider selected while adding the app.
So I think that field should not be allowed to be changed. How do you think?

Reply:
Actually I feel the same way. Currently changing it via the web client is allowed, but few users use it possibly.

Reply:
Applied at f05df12